connectors/aws-cert/src/main/java/com/vmware/connectors/aws/cert/AwsCertController.java

```diff
@@ -0,0 +1,331 @@
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.AsyncRestOperations;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+import rx.Observable;
+import rx.Single;
+
+import javax.validation.Valid;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RestController
+public class AwsCertController {
+
+    private static final String APPROVAL_URL_PARAM = "hero_aws_cert_approval_url";
+
+    private static final Logger logger = LoggerFactory.getLogger(AwsCertController.class);
+
+    private static final String ROUTING_PREFIX = "x-routing-prefix";
+
+    private static final String APPROVE_PATH = "/api/v1/approve";
+
+    private final String certificateApprovalHost;
+    private final String certificateApprovalPath;
+    private final AsyncRestOperations rest;
+    private final CardTextAccessor cardTextAccessor;
+
+    @Autowired
+    public AwsCertController(
+            @Value("${aws.certificate.connector.approval.host}") String certificateApprovalHost,
+            @Value("${aws.certificate.connector.approval.path}") String certificateApprovalPath,
+            AsyncRestOperations rest,
+            CardTextAccessor cardTextAccessor
+    ) {
+        this.certificateApprovalHost = certificateApprovalHost.toLowerCase(Locale.US);
+        this.certificateApprovalPath = certificateApprovalPath;
+        this.rest = rest;
+        this.cardTextAccessor = cardTextAccessor;
+    }
+
+    @PostMapping(
+            path = "/cards/requests",
+            produces = MediaType.APPLICATION_JSON_VALUE,
+            consumes = MediaType.APPLICATION_JSON_VALUE
+    )
+    public Single<ResponseEntity<Cards>> getCards(
+            @RequestHeader(ROUTING_PREFIX) String routingPrefix,
+            @Valid @RequestBody CardRequest request
+    ) {
+        logger.trace("getCards called, routingPrefix={}, request={}", routingPrefix, request);
+
+        List<String> approvalUrls = new ArrayList<>(request.getTokens("approval_urls"));
+
+        if (CollectionUtils.isEmpty(approvalUrls)) {
+            return Single.just(ResponseEntity.ok(new Cards()));
+        }
+
+        approvalUrls = validateUrls(approvalUrls);
+
+        Collections.sort(approvalUrls);
```

Comment:
I was having trouble with the order of the Set contents in the unit tests, so I added this sorting into a list then a LinkedHashSet. I'm not really happy about it. I'm open to suggestions.